packages/cysync-core/src/hooks/useButtonState.ts

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+export const useButtonState = (): [
+  boolean,
+  (shouldActivate: boolean) => void,
+] => {
+  const [btnState, setBtnState] = useState(false);
+
+  const handleButtonState = (shouldActivate: boolean) => {
+    setBtnState(shouldActivate);
+  };
+
+  return [btnState, handleButtonState];
+};
```

Comment:
Looks a lot like useState

Reply:
SInce we have used this in multiple files, we have created this hook.